### PR TITLE
SASE-19470: Update to use SearchID API to fetch data

### PR DESCRIPTION
**Pull request by atinm**
Change(s):
   - Update to use the latest SearchID based API to fetch data from Lookout backend servers

   Reason for Change(s):
   - APIs to the backend changed, breaking existing functionality

   Version Updated:
   - Updated to 3.0.0

   Testing Completed:
   - Validated in Microsoft Sentinel by installing as per guidance and seeing event fetching work
  
   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by atinm** [comment]
@microsoft-github-policy-service agree [company="Lookout, Inc."]

**Comment by atinm** [comment]
@microsoft-github-policy-service agree company="Lookout, Inc"

**Comment by v-rbajaj** [comment]
Hi @atinm, thanks for raising this PR, we will review this PR by 8 Nov 2023.

File: Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py [hunk 1]
```diff
@@ -59,188 +69,317 @@
...
-            past_time = (datetime.utcnow().replace(second=0, microsecond=0) - timedelta(days=int(pastDays))).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+            past_time = (
+                datetime.utcnow().replace(second=0, microsecond=0)
+                - timedelta(days=int(pastDays))
+            ).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
         return (past_time, current_time_day)
 
-    def get_Data(self,report_type_suffix,startTime,endTime):
-            
+    def get_SearchId(self, endpoint, report_type_suffix, startTime, endTime):
```
[the remark this answers is not in the document]
Comment (atinm) [reply Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py:124]: We expect to receive None on error, which is what is implict. We only return something in the explicit returns - the rest of the time we expect to return None. This is as designed.

File: Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py [hunk 2]
```diff
@@ -59,188 +69,317 @@
...
+                )
             else:
-                logging.error("Something wrong. Error code: {}".format(response.status_code))
+                logging.error(
+                    "Something wrong. Error code: {}".format(response.status_code)
+                )
         except Exception as err:
             logging.error("Something wrong. Exception error text: {}".format(err))
 
+    def get_Data(self, endpoint, searchId):
```
[the remark this answers is not in the document]
Comment (atinm) [reply Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py:166]: Same as above - we expect to return the implicit None when there is an error. Non-error cases have an explicit return. As designed.

File: Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py [hunk 3]
```diff
@@ -59,188 +69,315 @@
...
-            past_time = (datetime.utcnow().replace(second=0, microsecond=0) - timedelta(days=int(pastDays))).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+            past_time = (
+                datetime.utcnow().replace(second=0, microsecond=0)
+                - timedelta(days=int(pastDays))
+            ).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
         return (past_time, current_time_day)
 
-    def get_Data(self,report_type_suffix,startTime,endTime):
-            
+    def get_SearchId(self, endpoint, report_type_suffix, startTime, endTime):
```
[the remark this answers is not in the document]
Comment (atinm) [reply Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py:124]: Same as above

File: Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py [hunk 4]
```diff
@@ -59,188 +69,315 @@
...
+                )
             else:
-                logging.error("Something wrong. Error code: {}".format(response.status_code))
+                logging.error(
+                    "Something wrong. Error code: {}".format(response.status_code)
+                )
         except Exception as err:
             logging.error("Something wrong. Exception error text: {}".format(err))
 
+    def get_Data(self, endpoint, searchId):
```
[the remark this answers is not in the document]
Comment (atinm) [reply Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py:166]: Same as above

File: Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityViolations/__init__.py [hunk 5]
```diff
@@ -59,188 +69,315 @@
...
-            past_time = (datetime.utcnow().replace(second=0, microsecond=0) - timedelta(days=int(pastDays))).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+            past_time = (
+                datetime.utcnow().replace(second=0, microsecond=0)
+                - timedelta(days=int(pastDays))
+            ).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
         return (past_time, current_time_day)
 
-    def get_Data(self,report_type_suffix,startTime,endTime):
-            
+    def get_SearchId(self, endpoint, report_type_suffix, startTime, endTime):
```
[the remark this answers is not in the document]
Comment (atinm) [reply Solutions/Lookout Cloud Security Platform for Microsoft Sentinel/Data Connectors/LookoutCSConnector/LookoutCloudSecurityViolations/__init__.py:124]: Same as above

**Comment by v-rbajaj** [comment]
Hi @atinm, please repackage the solution using this tool -- > https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V3 as the provided maintemplate can't be deployed, after repackaging the issue should get resolved.

**Comment by atinm** [comment]
> Hi @atinm, please repackage the solution using this tool -- > https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V3 as the provided maintemplate can't be deployed, after repackaging the issue should get resolved.

@v-rbajaj When I run the v3, I get some api versions issues - but I am pushing up what seem to pass Test-AzTemplate

**Comment by v-rbajaj** [comment]
Hi @atinm, I deployed the maintemplate again from this PR and again the deployment failed. 

I repackaged this solution by pulling this PR on my local machine, there are some differences and to test the maintemplate I did deploy it and the deployment was successful. 
PFB screenshot.
image

I am uploading maintemplate here please push it on this PR and also update the 3.0.0 zip. 
[mainTemplate.json](https://github.com/Azure/Azure-Sentinel/files/13318063/mainTemplate.json)
[3.0.0.zip](https://github.com/Azure/Azure-Sentinel/files/13318064/3.0.0.zip)

**Comment by atinm** [comment]
> Hi @atinm, I deployed the maintemplate again from this PR and again the deployment failed.
> 
> I repackaged this solution by pulling this PR on my local machine, there are some differences and to test the maintemplate I did deploy it and the deployment was successful. PFB screenshot. image
> 

@v-rbajaj - the latest push is with the files you gave us, thank you for taking the time do this for us!

**Comment by atinm** [comment]
@v-rbajaj - added ReleaseNotes.md as per request. Thanks!

**Review by v-rbajaj — APPROVED** [review]

**Pull request merged**